runtime/op/ztests/user-join.yaml

```
@@ -0,0 +1,41 @@
+script: |
+  export ZED_LAKE=test
+  zed init -q
+  zed create -q -orderby flavor:asc fruit
+  zed create -q -orderby likes:asc people
+  zed load -q -use fruit fruit.zson
+  zed load -q -use people people.zson
+  zed query -z -I test.zed
+
+inputs:
+  - name: test.zed
+    data: |
+      op test(lpool, rpool, lkey, rkey, ldest, rsrc): (
+        from lpool
+        | inner join ( from rpool ) on lkey = rkey ldest := rsrc
+      )
+      test("fruit", "people", flavor, likes, eater, name)
```

Maybe simplify by joining files instead of pools?
```suggestion
  zq -z -I test.zed

inputs:
  - name: test.zed
    data: |
      op test(lfile, rfile, lkey, rkey, ldest, rsrc): (
        file lfile
        | inner join ( file rfile ) on lkey = rkey ldest := rsrc
      )
      test("fruit.zson", "people.zson", flavor, likes, eater, name)
```

---

This is testing an issue that phil found where things were working with files but not for pools.